presto-main/src/main/java/com/facebook/presto/operator/PagesRTreeIndex.java

```diff
@@ -206,6 +213,15 @@ public boolean isJoinPositionEligible(int joinPosition, int probePosition, Page
         return filterFunction == null || filterFunction.filter(joinPosition, probePosition, probe);
     }
 
+    private void maybeAccelerateGeometry(OGCGeometry ogcGeometry) {
+        if (radiusChannel == -1 && ogcGeometry.geometryType() != OGCConcreteGeometryCollection.TYPE) {
+            // If radiusChannel is supplied, this is a distance query, for which our acceleration won't help.
+            // Acceleration fails for GeometryCollections, so don't try those.
```

Comment:
@jagill The following should work for geometry collections and regular geometries:

```
            GeometryCursor cursor = geometry.getEsriGeometryCursor();
            while (true) {
                com.esri.core.geometry.Geometry esriGeometry = cursor.next();
                if (esriGeometry == null) {
                    break;
                }

                relateOperator.accelerateGeometry(esriGeometry, null, Geometry.GeometryAccelerationDegree.enumMild);
            }
```

Reply:
But couldn't the geometry collection contain elements that are geometry collections?

Reply:
@tdcmeehan It could, but `getEsriGeometryCursor` iterates over flattened collection.

https://github.com/Esri/geometry-api-java/blob/master/src/main/java/com/esri/core/geometry/ogc/OGCConcreteGeometryCollection.java#L356